packages/block-editor/src/components/url-input/index.js

```diff
@@ -22,229 +22,211 @@ import { withInstanceId, withSafeTimeout, compose } from '@wordpress/compose';
-				// reset or deleted on subsequent requests or unmounting.
-				if ( this.suggestionsRequest !== request ) {
-					return;
-				}
+	const updateSuggestions = debounce( async ( search = '' ) => {
+		setSelectedSuggestion( null );
 
-				this.setState( {
-					suggestions,
-					loading: false,
-					showSuggestions: !! suggestions.length,
-				} );
-
-				if ( !! suggestions.length ) {
-					this.props.debouncedSpeak(
-						sprintf(
-							/* translators: %s: number of results. */
-							_n(
-								'%d result found, use up and down arrow keys to navigate.',
-								'%d results found, use up and down arrow keys to navigate.',
-								suggestions.length
-							),
-							suggestions.length
-						),
-						'assertive'
-					);
-				} else {
-					this.props.debouncedSpeak(
-						__( 'No results.' ),
-						'assertive'
-					);
-				}
-				this.isUpdatingSuggestions = false;
-			} )
-			.catch( () => {
-				if ( this.suggestionsRequest === request ) {
-					this.setState( {
-						loading: false,
-					} );
-					this.isUpdatingSuggestions = false;
-				}
-			} );
+		if ( ! allowSuggestionsRequest( search ) ) {
+			setShowSuggestions( false );
+			setLoading( false );
+			return;
+		}
+		setLoading( true );
 
-		// Note that this assignment is handled *before* the async search request
-		// as a Promise always resolves on the next tick of the event loop.
-		this.suggestionsRequest = request;
-	}
+		const request = fetchLinkSuggestions( search, {
+			isInitialSuggestions: isEmpty( search ),
+		} );
+		pendingRequest.current = request;
+
+		try {
+			const result = await request;
+			if ( result && request === pendingRequest.current ) {
+				setSuggestions( result );
+				setShowSuggestions( !! result.length );
+				speakUpdateResult();
+				clearUpdateSuggestions();
+			}
+		} catch ( _error ) {
+			// Release updating state if the current request fails.
+			if ( pendingRequest.current === request ) {
+				clearUpdateSuggestions();
+			}
+		}
+	}, REQUEST_DEBOUNCE_DELAY );
 
-	onChange( event ) {
+	function onChange( event ) {
 		const inputValue = event.target.value;
 
-		this.props.onChange( inputValue );
-		if ( ! this.props.disableSuggestions ) {
-			this.updateSuggestions( inputValue.trim() );
+		parentOnChange( inputValue );
+		if ( ! disableSuggestions ) {
+			updateSuggestions( inputValue.trim() );
 		}
 	}
 
-	onFocus() {
-		const { suggestions } = this.state;
-		const { disableSuggestions, value } = this.props;
-
+	function onFocus() {
 		// When opening the link editor, if there's a value present, we want to load the suggestions pane with the results for this input search value
 		// Don't re-run the suggestions on focus if there are already suggestions present (prevents searching again when tabbing between the input and buttons)
 		if (
 			value &&
 			! disableSuggestions &&
-			! this.isUpdatingSuggestions &&
-			! ( suggestions && suggestions.length )
+			! isUpdatingSuggestions() &&
+			isEmpty( suggestions )
```

Comment:
Nitpick again - why do you think we need to use lodash `isEmpty`?
In many cases we check for a `! string` or `! array?.length`.

Reply:
Updated in f6a9403